themes/default/files/scripts/exam-display.js

```diff
@@ -294,14 +301,49 @@ Numbas.queueScript('exam-display',['display-base','math','util','timing'],functi
          */
         this.enteredPassword = Knockout.observable('');
 
-        /** Can the exam begin? True if no password is required, or if the student has entered the right password.
+        /** Does this exam allow the student to download their attempt data?
+         *
+         * @member {boolean} allowAttemptDownload
+         * @memberof Numbas.display.ExamDisplay
+         */
+        this.allowAttemptDownload = e.settings.allowAttemptDownload;
+
+        /** Must the student write their name before the exam can begin?
+         *
+         * @member {boolean} needsStudentName
+         * @memberof Numbas.display.ExamDisplay
+         */
+        this.needsStudentName = this.allowAttemptDownload;
```

Comment:
needsStudentName exam setting has been removed

Reply:
I kept the display property in case there's ever another reason to ask for the student's name. It's defined to be true iff `allowAttemptDownload` is true at the moment.